specification/specs/indices/mapping_management/put_mapping/PutMappingRequest.ts

```diff
@@ -47,7 +48,7 @@ interface PutMappingRequest extends RequestBase {
     index_field?: IndexField
     meta?: Dictionary<string, UserDefinedValue>
     numeric_detection?: boolean
-    properties?: Dictionary<PropertyName, PropertyBase>
+    properties?: Dictionary<PropertyName, Property>
```

Comment:
Fixing the properties types is a beast. @swallez and me are working on the `variants` feature of the compiler to improve the definitions here. As is, this API can't be fixed until we finish the `variants` feature.